static/js/components/RepeatableContentListing.test.tsx

```diff
@@ -254,6 +255,31 @@ describe("RepeatableContentListing", () => {
     ).toBe(false)
   })
 
+  test("should filter based on query param", async () => {
+    helper.browserHistory.push("/?q=search")
+    const { wrapper } = await render()
+    contentListingItems.forEach((item, idx) => {
+      const li = wrapper.find("li").at(idx)
+      expect(li.text()).toContain(item.title)
+    })
+  })
+
+  test("should let the user filter via text input", async () => {
+    const { wrapper } = await render()
+    const filterInput = wrapper.find(".site-search-input")
+    const event = {
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      preventDefault() {},
+      target: { value: "my-search-string" }
+    } as React.ChangeEvent<HTMLInputElement>
+    filterInput.simulate("change", event)
+    await wait(800)
+    wrapper.update()
+    expect(wrapper.find(".site-search-input").prop("value")).toBe(
+      "my-search-string"
+    )
```

Comment:
**Question / Requested Change**: What's the goal of this test? I think it might not be doing what is expected. 

Doesn't the input update immediately, but the history / url bar has debounced updating? I tried this test out and the assertion passes without the `wait(800)`.

Is the goal to test the debouncing history thing? If yes, two suggestions:
- I think `jest.spyOn(helper.browserHistory, 'push')` should let us spy on the url bar... but maybe theres a better way. Unsure.
- Instead of using `wait`, you could use `jest.useFakeTimers` and `jest.advanceFakeTimers` ... :timelord:

    Although, I imagine `wait` isn't a significant contributor to our test runtime 🤷 

Anyway, requested change is: remove the wait(800) if the test is the way it is is supposed to be, or make it fail without the wait(800).

Reply:
I had a heck of a time with this test! I'll give `spyOn` a try, but it was the strangest thing - although I had an almost identical test to the corresponding one for the `SitesDashboard`, for some reason the calls to `history.push` in the hook weren't changing any kind of history anywhere that I could pull out and assert, so I got a little frustrated and then just asserted that the new search string was present as the `value` prop on the input. I'll give it another go today :)

Reply:
ok `spyOn` worked, but still if I `console.log(helper.browserHistory.location)` the `.search` prop is `""`... 🤷‍♀️